File: MeasureUUUP/src/main/java/com/thejakeofink/measureup/MainActivity.java
package com.thejakeofink.measureup;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.View;

public class MainActivity extends Activity {
    private final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.main, menu);
        return true;
    }

    public void runLength(View v)
    {
        /*Button button = (Button)findViewById(R.id.length_button);
        TextView textView = (TextView)findViewById(R.id.test_textview);
        String text = button.getText().toString();
        textView.setText(text);*/

        Intent intent = new Intent(this, LengthActivity.class);
        startActivity(intent);
    }

    public void runVolume(View v)
    {
        /*Button button = (Button)findViewById(R.id.volume_button);
        TextView textView = (TextView)findViewById(R.id.test_textview);
        String text = button.getText().toString();
        textView.setText(text);*/

        Intent intent = new Intent(this, VolumeActivity.class);
        startActivity(intent);
    }

    public void runTemp(View v)
    {
        /*Button button = (Button)findViewById(R.id.temp_button);
        TextView textView = (TextView)findViewById(R.id.test_textview);
        String text = button.getText().toString();
        textView.setText(text);*/

        Intent intent = new Intent(this, TempActivity.class);
        startActivity(intent);
    }

    public void runMass(View v)
    {
        /*Button button = (Button)findViewById(R.id.mass_button);
        TextView textView = (TextView)findViewById(R.id.test_textview);
        String text = button.getText().toString();
        textView.setText(text);*/

        Intent intent = new Intent(this, MassActivity.class);
        startActivity(intent);
    }

    public void runCurrency(View v)
    {
        /*Button button = (Button)findViewById(R.id.currency_button);
        TextView textView = (TextView)findViewById(R.id.test_textview);
        String text = button.getText().toString();
        textView.setText(text);*/

        Intent intent = new Intent(this, CurrencyActivity.class);
        startActivity(intent);
    }

    public void runData(View v)
    {
        /*Button button = (Button)findViewById(R.id.data_button);
        TextView textView = (TextView)findViewById(R.id.test_textview);
        String text = button.getText().toString();
        textView.setText(text);*/

        Intent intent = new Intent(this, DataActivity.class);
        startActivity(intent);
    }

    public void runTip(View v)
    {
        Intent intent = new Intent(this, TipActivity.class);
        startActivity(intent);
    }

}

File: MeasureUUUP/src/main/java/com/thejakeofink/measureup/MainActivity.kt
package com.thejakeofink.measureup

import android.os.Bundle
import android.support.design.widget.NavigationView
import android.support.design.widget.Snackbar
import android.support.v4.widget.DrawerLayout
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    private val TAG = "MainActivity"

    private var drawer: DrawerLayout? = null
    private var toolbar: Toolbar? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        toolbar = findViewById(R.id.main_toolbar) as Toolbar?

        setSupportActionBar(toolbar)

        val actionBar = supportActionBar

        actionBar.title = "We did it!"

        drawer = findViewById(R.id.drawer_view) as DrawerLayout?

        var drawerToggle = object : ActionBarDrawerToggle(this, drawer, toolbar, R.string.empty, R.string.empty) {

            override fun onDrawerOpened(drawerView: View?) {
                super.onDrawerOpened(drawerView)
            }

            override fun onDrawerClosed(drawerView: View?) {
                super.onDrawerClosed(drawerView)
            }
        }

        drawer?.setDrawerListener(drawerToggle);
        drawerToggle.syncState()

        val navView = findViewById(R.id.nav_view) as NavigationView

        navView.setNavigationItemSelectedListener {
            var toastText = getStringData(it.itemId)
            Snackbar.make(drawer, toastText, Snackbar.LENGTH_INDEFINITE).show()
            return@setNavigationItemSelectedListener true
        }
    }

    fun getStringData(action: Int): String {
        when (action) {
            R.id.action_data -> return getString(R.string.title_activity_data)
            R.id.action_length -> return getString(R.string.title_activity_length)
            R.id.action_mass -> return getString(R.string.title_activity_mass)
            R.id.action_temp -> return getString(R.string.title_activity_temp)
            R.id.action_tip -> return getString(R.string.title_activity_tip)
            R.id.action_volume -> return getString(R.string.title_activity_volume)
            else -> return ""
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {

        when (item?.itemId) {
            R.id.action_settings -> {
                Snackbar.make(drawer, "Some Snacky Snack", Snackbar.LENGTH_INDEFINITE).show()
            }
        }

        return super.onOptionsItemSelected(item)
    }
}
